ciao-controller/internal/datastore/datastore.go

```diff
@@ -1490,9 +1489,45 @@ func (ds *Datastore) GetCNCIWorkloadID() (string, error) {
 
 // GetNodeSummary provides a summary the state and count of instances running per node.
 func (ds *Datastore) GetNodeSummary() ([]*types.NodeSummary, error) {
-	// TBD: write a new routine that grabs the node summary info
-	// from the cache rather than do this lengthy sql query.
-	return ds.db.getNodeSummary()
+	var nodes []*types.NodeSummary
+
+	ds.nodesLock.RLock()
+
+	for _, n := range ds.nodes {
+		var total []*types.Instance
+		var running []*types.Instance
+		var pending []*types.Instance
+		var paused []*types.Instance
```

Comment:
As we don't actually care about the instances themselves can we we just use some ints here? Or even better just increment the values in the summary struct:

```
 summary.TotalInstances++
 switch i.State {
 	case payloads.Pending:
 	        // Do nothing?
 	case payloads.Running:
 		summary.TotalRunnningInstances++
 	case payloads.Exited:
 		summary.TotalPausedInstances++
}
```
WDYT?